src/model.jl

```diff
@@ -1003,20 +1010,24 @@ function constraint(ocp::OptimalControlModel{T, V}, label::Symbol) where {T <: T
         (:boundary, f::BoundaryConstraint, _, _) => return f
         (:control , f::ControlConstraint,  _, _) => return f
         (:control , rg,   _, _) => begin
-            C = nothing
-            (T, V) == (Autonomous, Fixed) && (C = ControlConstraint(u         -> u[rg], T, V))
-            (T, V) == (Autonomous, NonFixed)   && (C = ControlConstraint((u, v)    -> u[rg], T, V))
-            (T, V) == (NonAutonomous, Fixed)   && (C = ControlConstraint((t, u)    -> u[rg], T, V))
-            (T, V) == (NonAutonomous, NonFixed)     && (C = ControlConstraint((t, u, v) -> u[rg], T, V))
+            C = @match ocp begin
+                ::OptimalControlModel{Autonomous, Fixed} => ControlConstraint(u         -> u[rg], T, V)
+                ::OptimalControlModel{Autonomous, NonFixed} => ControlConstraint((u, v)    -> u[rg], T, V)
+                ::OptimalControlModel{NonAutonomous, Fixed} => ControlConstraint((t, u)    -> u[rg], T, V)
+                ::OptimalControlModel{NonAutonomous, NonFixed} => ControlConstraint((t, u, v) -> u[rg], T, V)
+                _ => nothing
```

Comment:
@BaptisteCbl for this last match pattern (and other similar cases in the code), I am not sure that returning `nothing` makes sense. As matching must be exhaustive, it is probably OK to assume that the last case has to be the only possible one.